server/routes_test.go

```diff
@@ -106,11 +106,11 @@ func TestServerRoutesWithAuthAndBCrypt(t *testing.T) {
 	srvB := RunServer(optsB)
 	defer srvB.Shutdown()
 
-	urlA := fmt.Sprintf("nats://%s:%d/", optsA.Host, optsA.Port)
-	urlB := fmt.Sprintf("nats://%s:%d/", optsB.Host, optsB.Port)
+	urlA := fmt.Sprintf("nats://%s:%s@%s:%d/", optsA.Username, optsA.Password, optsA.Host, optsA.Port)
+	urlB := fmt.Sprintf("nats://%s:%s@%s:%d/", optsB.Username, optsB.Password, optsB.Host, optsB.Port)
 
 	// Wait for route to form.
-	time.Sleep(250 * time.Millisecond)
+	time.Sleep(4 * time.Second)
```

Comment:
If we feel that 5 seconds may not always be enough we can increase. Again, the function may return before, so if setting to 10 avoid a flapping test, I think it is worth bumping it.